clang/unittests/Analysis/LifetimeSafetyTest.cpp

```diff
@@ -1202,5 +1202,241 @@ TEST_F(LifetimeAnalysisTest, LivenessOutsideLoop) {
   EXPECT_THAT(Origins({"p"}), MaybeLiveAt("p1"));
 }
 
```

Comment:
Meta comment for this file:
I do not think the new tests test the new changes in this PR.
We would need to test the state at after the return statement or at the end of the block. Maybe we can expose this information with a test-only API in lifetime safety.

Reply:
Yeah but checking before the return statement does not test for the changes done in this PR. These tests should pass already before this PR.